edit_course/views.py

```diff
@@ -122,6 +124,22 @@ def renumber_recursion(self, parent, n=1):
             n += 1
         return n
 
+    def number_learning_objects_consecutively(self) -> None:
+        """Number all learning objects (i.e. chapters and exercises)"""
+        def inner(parent: Union[CourseModule, CourseChapter], n: int):
+            for entry in parent['children']:
+                model = LearningObject.objects.get(id=entry['id'])
+                model.consecutive_order = n
+                model.save()
+                n += 1
+                if not model.is_submittable:
+                    n = inner(entry, n)
+            return n
+
+        n = 1
+        for module in self.content.modules():
+            n = inner(module, n)
```

Comment:
General comment about cache (`self.content` is a CachedContent) usage: the idea behind it is to avoid querying the database. If you are going to query the database anyway and not use any precomputed values within the cache, there is no point in using it.

The above function is said to take a `CourseModule` but `module` here isn't actually a `CourseModule` object but a dict. The `get_common_objects` method above should set self.modules to be the CourseModule objects, so you can use that instead.

Reply:
If the order is calculated dynamically, you can just use the cache directly, so there are no need to use model objects here.